### DISCUSSION TITLE
using https or ssl for localhost development environment

### DISCUSSION THREAD

**Discussion by Sutikshan** [jaredpalmer/razzle#1484] [Q&A] [answered]
Hello All,
I am trying to configure razzle via razzle.config.js to kick start development server on `https` protocol. For this I created encryption key following this gist - https://gist.github.com/pgilad/63ddb94e0691eebd502deee207ff62bd

Now I am trying to config `razzle.config.js` to start in https on local dev env - 
```
module.exports = {
  modifyWebpackConfig({
    env: {
      target, // the target 'node' or 'web'
      dev, // is this a development build? true or false
    },
    webpackConfig,
  }) {
    if (target === 'web' && dev) {
      return {
        ...webpackConfig,
        devServer: {
          ...webpackConfig.devServer,
          http2: true,
          https: {
              key: fs.readFileSync('./private.key'),
              cert: fs.readFileSync('./private.pem'),
          }
      }
      };

    }

    return webpackConfig;
  }
};

```
What else I am suppose to do?
Is there anything wrong in above config?
Chrome shows `ERR_SSL_PROTOCOL_ERROR`
Do I need to create server using `https` module?

**Comment by fivethreeo** [comment]
Yes you need to use https in server.js

**Reply by Sutikshan** [reply]
Thanks, I removed the config and resorted to 
```
if (environmentName === 'local') {
  server = https.createServer(
    {
      key: fs.readFileSync('./private.key'),
      cert: fs.readFileSync('./private.pem'),
    },
    app
  );
} else {
  server = http.createServer(app);
}
```
which worked for me.

**Reply by rarescosmin** [reply]
hi @Sutikshan could you please share your full server.js file please? 💯

**Comment by therealgilles** [comment]
An alternative in development is to use a proxy so that you have https on the client side but http on the server. This is useful for me as my app is deployed behind a proxy server.

**Comment by fivethreeo** [comment, marked as answer]
There is a example that uses devcert to use ssl in local dev.

https://github.com/jaredpalmer/razzle/tree/master/examples/with-devcert-https
